### PR TITLE
Withdrawal deals sanity damage, tweak perks focused on addiction

### PR DESCRIPTION
**Pull request by Kegdo**
## About The Pull Request
alcoholic fate perk gets permanent addiction, alcoholic gets bonus of 15 stats, and yes this is a fair and balanced tradeoff
drug addict now gets a bottle of 12 pills full of 15 units of a random stim they are addicted to; before, they got FUCKING __60__(!!!) units of a drug in one pill, AND TWO BOTTLES of said pills(~25). also permanent addiction yea
withdrawal deals gradually more sanity damage when you get the... announcement? english hard
anyways: 
stage1 - 0 (sanity dmg)
stage2 - 0
stage3 - 5
stage4 - 10
end - +10 sanity restoration
## Why It's Good For The Game
trillionardos of people complained that withdrawal does nothing(except for stat debuffs for certain chems)
now, in REAL WORLD(sic!) withdrawals are pain!! time to add realism
also fates focused on addiction are actually addicted and good now, with an interesting tradeoff
## Changelog
:cl:
add: sanity damage on withdrawal
tweak: drug addict does not get two bottles of 60u pills now
tweak: addiction fate perks have permanent addiction
tweak: increased alcoholic bonus
/:cl:

### CODE REVIEW

File: code/modules/reagents/metabolism.dm [hunk 1]
```diff
@@ -207,19 +207,20 @@
...
-					R.addiction_act_stage3(parent)
+					R.addiction_act_stage2(parent)
 				if(30 to 40)
+					R.addiction_act_stage3(parent)
+				if(40 to 50)
+					R.addiction_act_stage4(parent)
+				if(50 to INFINITY && ((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim))))
 					R.addiction_act_stage4(parent)
-				if(40 to INFINITY)
+				if(50 to INFINITY && !((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim))))
```
Comment (SirRichardFrancis) [inline code/modules/reagents/metabolism.dm:223]:
```suggestion
				else
```

File: code/modules/reagents/reagents.dm [hunk 2]
```diff
@@ -215,24 +215,28 @@
 	return null
 
 // Addiction
-/datum/reagent/proc/addiction_act_stage1(mob/living/carbon/M)
-	if(prob(30))
+// Addiction
+/datum/reagent/proc/addiction_act_stage1(mob/living/carbon/human/M)
+	if(prob(30) && ishuman(M))
```
Comment (SirRichardFrancis) [inline code/modules/reagents/reagents.dm:220]:
```suggestion
	if(prob(30))
```

File: code/modules/reagents/reagents.dm [hunk 3]
```diff
@@ -215,24 +215,28 @@
...
+/datum/reagent/proc/addiction_act_stage3(mob/living/carbon/human/M)
 	if(prob(30))
 		to_chat(M, SPAN_DANGER("You have an intense craving for [name]."))
+		M.sanity.changeLevel(-5)
 
-/datum/reagent/proc/addiction_act_stage4(mob/living/carbon/M)
+/datum/reagent/proc/addiction_act_stage4(mob/living/carbon/human/M)
 	if(prob(30))
 		to_chat(M, SPAN_DANGER("You're not feeling good at all! You really need some [name]."))
+		M.sanity.changeLevel(-10)
```
Comment (SirRichardFrancis) [inline code/modules/reagents/reagents.dm:235]: That may (and very likely will) cause people to go insane if they forgot to put another cig in their mouth after old one is gone. Only annoyance.

File: code/modules/reagents/metabolism.dm (hunk 1 above)
Comment (Kegdo) [reply code/modules/reagents/metabolism.dm:223]: ELSE DOES NOT WORK i tried unfortunately
that else checks for every IF above, so, for example, if you are at stage 1, you aren't at stage 2(duh), which makes the 'else' proc

File: code/modules/reagents/reagents.dm (hunk 3 above)
Comment (Kegdo) [reply code/modules/reagents/reagents.dm:235]: maybe make nicotime unaddictable?

Comment (Kegdo) [reply code/modules/reagents/reagents.dm:235]: image
no we cant do it now

File: code/modules/reagents/metabolism.dm (hunk 1 above)
Comment (SirRichardFrancis) [reply code/modules/reagents/metabolism.dm:223]: Bruh. Okay, try this then.
```suggestion
				if(50 to INFINITY)
					if((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim)))
						R.addiction_act_stage4(parent)
					else
						R.addiction_end(parent)
						addiction_list.Remove(R)
```

Comment (Kegdo) [reply code/modules/reagents/metabolism.dm:223]: yeh should wok

File: code/modules/reagents/reagents.dm (hunk 3 above)
Comment (Kegdo) [reply code/modules/reagents/reagents.dm:235]: > > maybe make nicotime unaddictable?
> 
> Nah, nicotine is very addictive.

then its intended; yet another smoking nerf!

**Comment by SirRichardFrancis** [comment]
Also on the topic on late stage of addiction damaging sanity. People with "addict" and "alcoholic" fates might find themselves in a state of constant breakdown with no possibility of removing addiction. No fun.

**Comment by Voldirs** [comment]
disco-elysium-smile

**Comment by TheShown911** [comment]
> Also on the topic on late stage of addiction damaging sanity. People with "addict" and "alcoholic" fates might find themselves in a state of constant breakdown with no possibility of removing addiction. No fun.

Immersive.

File: code/modules/reagents/metabolism.dm [hunk 4]
```diff
@@ -207,21 +207,23 @@
...
 				if(30 to 40)
+					R.addiction_act_stage3(parent)
+				if(40 to 50)
 					R.addiction_act_stage4(parent)
-				if(40 to INFINITY)
-					R.addiction_end(parent)
-					addiction_list.Remove(R)
+				if(50 to INFINITY)
+					if((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim)))
+						R.addiction_act_stage4(parent)
```
Comment (TheShown911) [inline code/modules/reagents/metabolism.dm:223]: Maybe make it cycle back to the first addiction stage so it loops.

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]
My thoughts.

**Review by Shadow-Quill — COMMENTED** [review]
Make sure you update your master branch, then merge it into this PR's.

File: code/modules/reagents/metabolism.dm (hunk 4 above)
Comment (Kegdo) [reply code/modules/reagents/metabolism.dm:223]: why? it doesn't make sense

**Comment by Kegdo** [comment]
how do i aprove my own pr

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**